django/db/models/functions/base.py

```diff
@@ -203,8 +203,7 @@ def __init__(self, expression, substring, **extra):
         """
         if not hasattr(substring, 'resolve_expression'):
             substring = Value(substring)
```

Comment:
Are you planning to remove these lines in the same PR? It looks to me that this whole `__init__` could be replaced by a simple `_output_field = fields.IntegerField()`.